website/common/script/ops/buy/buySpecialSpell.js

```diff
@@ -9,19 +9,34 @@ import {
   NotFound,
 } from '../../libs/errors';
 
-module.exports = function buySpecialSpell (user, req = {}) {
+module.exports = function buySpecialSpell (user, req = {}, analytics) {
   let key = get(req, 'params.key');
+  let quantity = req.quantity || 1;
+
   if (!key) throw new BadRequest(i18n.t('missingKeyParam', req.language));
 
   let item = content.special[key];
   if (!item) throw new NotFound(i18n.t('spellNotFound', {spellId: key}, req.language));
 
-  if (user.stats.gp < item.value) {
+  if (user.stats.gp < item.value * quantity) {
     throw new NotAuthorized(i18n.t('messageNotEnoughGold', req.language));
   }
-  user.stats.gp -= item.value;
+  user.stats.gp -= item.value * quantity;
```

Comment:
Are you removing the loop? I'd rather not have the quantity code in multiple places

Reply:
yes I removed the loop, I think its much cleaner having the `quantity` inside of each op, instead of having a loop which calls the method per iteration. Its also the better way for performance

Reply:
Yea, I can see that. But, this does show a lot of duplication and now is a case of "Shotgun Surgery." So, it would be good to remove that.

Reply:
What do you think about a shared method of checking the currency by type, value and quantity and the same for removing the currency ?

Reply:
You can check by type, but I think what we should have is an object or class with the value method. Then using one of the various methods of Javascript inheritance, we can return value *quantity on a `getValue` method and subtract that from a user in the buy method. Is the what you are also thinking?

Reply:
yes that is planned too :), I just wanted to have the first refactoring to be api save, once thats ok, the inner refactoring starts in a separate PR

Reply:
Yeah you are right, gonna focus on the coverage first